File: server/app/Model/UserAccountLog.php
<?php

declare(strict_types=1);

namespace App\Model;

use Hyperf\DbConnection\Model\Model;

/**
 * @property int $id
 * @property int $user_id 用户id
 * @property int $type 类型 1:登录,2:注册,3:重置密码,4:绑定手机,5:绑定邮箱,6:解绑手机,7:解绑邮箱,8:禁用账户,9:删除账户,10:绑定2fa,11:解绑2fa
 * @property string $ip
 * @property string $os 操作系统
 * @property string $device_id 设备唯一标识（iOS/Android/Web通用）
 * @property string $country_code 国家代码
 * @property string $country 国家
 * @property string $region 省
 * @property string $city 市
 * @property string $created_at
 */
class UserAccountLog extends Model
{
    public const UPDATED_AT = null;
    /**
     * The table associated with the model.
     */
    protected ?string $table = 'user_account_log';

    /**
     * The attributes that are mass assignable.
     */
    protected array $fillable = ['id', 'user_id', 'type', 'ip', 'os', 'device_id', 'country_code', 'country', 'region', 'city', 'created_at'];

    /**
     * The attributes that should be cast to native types.
     */
    protected array $casts = ['id' => 'integer', 'user_id' => 'integer', 'type' => 'integer'];
}

File: server_go/internal/app/model/user_account_log.go
package model

import "time"

// UserAccountLog 对应 user_account_log
type UserAccountLog struct {
	ID          int64     `gorm:"primaryKey"`
	UserID      int64     `gorm:"column:user_id"`
	Type        int       `gorm:"column:type"`
	IP          string    `gorm:"size:64"`
	OS          string    `gorm:"size:64"`
	DeviceID    string    `gorm:"column:device_id;size:128"`
	CountryCode string    `gorm:"column:country_code;size:16"`
	Country     string    `gorm:"size:64"`
	Region      string    `gorm:"size:64"`
	City        string    `gorm:"size:64"`
	CreatedAt   time.Time `gorm:"column:created_at"`
}

func (UserAccountLog) TableName() string {
	return P + "user_account_log"
}
